theseus_gui/src/helpers/settings.js

```diff
@@ -50,3 +50,16 @@ export async function get() {
 export async function set(settings) {
   return await invoke('settings_set', { settings })
 }
+
+/*
+Non-Tauri helpers
+*/
+
+export function deepEqual(x, y) {
+  const ok = Object.keys,
+    tx = typeof x,
+    ty = typeof y
+  return x && y && tx === 'object' && tx === ty
+    ? ok(x).length === ok(y).length && ok(x).every((key) => deepEqual(x[key], y[key]))
+    : x === y
+}
```

Comment:
this won't be needed with future changes

Reply:
What future changes is thsi waiting on?